ir/memory.cpp

```diff
@@ -149,6 +149,14 @@ void Pointer::is_dereferenceable(unsigned bytes, unsigned align) {
   is_dereferenceable(expr::mkUInt(bytes, m.bits_for_offset), align);
 }
 
+expr disjoint(expr offset1, const expr &len1, expr offset2, const expr &len2) {
+  return ((offset1+len1).ule(offset2) || (offset2+len2).ule(offset1));
```

Comment:
Is this condition correct if offset + len overflows?  Or can't that happen (because of the half of the address space limitation)?  If that's the case, we need a comment there explaining the assumption/reasoning.

Reply:
When I use is_disjoint in memcpy, overflow checking is run in is_dereferenceable.

If we need this condition at other code, overflow condition is considered.

I add a comment in the code also.